File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/MotorStatusCommand.java
package wpam.mobile_client.protocol;
import java.util.ArrayList;

public final class MotorStatusCommand extends Command
{
	public MotorStatusCommand()
	{
		super(10, CommandType.MOTOR_STATUS_COMMAND);
	}
	public final void close()
	{
		super.close();
	}

	@Override
	public ArrayList<Integer> getFrameBytes()
	{
		initializeDataSize();

		ArrayList<Integer> frame = getHeader();
		frame.add(commandType_.getValue());

		return frame;
	}
	@Override
	public String getName()
	{
		return new String("MotorStatusCommand");
	}
	@Override
	public void accept(CommandVisitor visitor)
	{
		visitor.visit(this);
	}

	@Override
	protected void initializeDataSize()
	{
		setDataSize(1);
	}
}

File: server/src/protocol/MotorStatusCommand.cpp
#include "MotorStatusCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

MotorStatusCommand::MotorStatusCommand()
        : Command(10, CommandType::MOTOR_STATUS_COMMAND)
{}

MotorStatusCommand::~MotorStatusCommand()
{}

vector<uint8_t> MotorStatusCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string MotorStatusCommand::getName()
{
    return string("MotorStatusCommand");
}

void MotorStatusCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void MotorStatusCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}
